### ISSUE TITLE
Inconsistant behaviour 

### ISSUE THREAD

**Issue by anasyb** [balderdashy/waterline#1177]
I have a array of IDs, for example 3 of them. When calling `User.find(IDs).exec(console.log)` I get the three records.
When I call `User.find().where(IDs).exec(console.log)` I get all records.
I have to do `User.find().where({id: IDs}).exec(console.log)` to get the ones I wanted

Moreover, when calling `User.find().where(IDs[0]).exec(console.log)` I get the correct one.

Very inconsistant, perhaps a bug with calling .where with an array

**Comment by particlebanana** [comment]
It should always be called with `{ id: IDs }`. Passing a primary key into a query is only a valid query on a `findOne` method. This should return an error.

**Comment by anasyb** [comment]
so only `.findOne`... then I agree, the rest should all throw errors as you said... including `.find()`, `where` which are working fine now if passed an array of IDs (at least with mongo)

**Comment by sailsbot** [comment]
Thanks for posting, @anasyb.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!

**Comment by devinivy** [comment]
Marked as a bug, so keeping this open.
